### ISSUE TITLE
Tag management/bulk editing

### ISSUE THREAD

**Issue by filcuk** [almarklein/timetagger#366]
The (recent?) addition of tag renaming is fantastic.

I would like to have an ability to select multiple tasks at once to add/remove tags in bulk.
Likewise, it would be great being able to use the Totals panel to add/remove tags for any tasks affected/in view.

Even if this isn't somethign you may want to add, the tool has been invaluable to me lately and I'm very happy with it.

**Comment by almarklein** [comment]
Thanks for the suggestion. At the moment I don't know how/where this functionality would fit in, but it sounds like a very valuable improvement!

**Comment by Tagirijus** [comment]
Instead of opening another issue, I extend this issue by commenting:

I'd find a _bulk delete_ of **tasks** (not only tags) useful. At the moment you can bulk delete tags. If you do so until the recorded tasks have no tags anymore, they get some "un-categorized" tag or so automatically. Maybe it would be a great idea to let the app delete those tasks instead. This way such a bulk-delete could be implemented already without much effort, I guess.

**Comment by Tagirijus** [comment]
Another idea came up: with a **tag manager** it would be cool to be able to import and export tags. E.g. when you already colorized and prioritized tags - or if you want to do so before you even track such tags.

Maybe there could just be a dedicated tag manager for all these ideas in this issue? Accessible through the menu, or so.

**Comment by almarklein** [comment]
> Maybe it would be a great idea to let the app delete those tasks instead.

No, records without tags are fine. It would be weird to auto-delete them. Some people don't use tags, or not always.

**Comment by Tagirijus** [comment]
> No, records without tags are fine. It would be weird to auto-delete them. Some people don't use tags, or not always.

Hm, I see. Maybe with an option, the user could set up then. Just a rough idea. The problem for me right now would be: I cannot delete tracked entries in a bulk. Thus my idea so that such entries would not get "uncategorized" but get deleted completely. Maybe even an option during the tag-renaming. Like having a checkbox "delete untagged entries, instead of givin #uncategorized" or so.

The other solution would be a proper bulk editor, but I guess this is far more complicated, for sure.

**Comment by kuon** [comment]
I would love to be able to bulk edit tasks too. My use case is when I export a report, I want to make the tasks as "billed".

There are three places where adding tag could be inserted in the UI (I have a preference for the third one):

- When you pick a time range, on the right you have a list of time entries arranged by tag, each row could have a + button on hover to add a tag
- When you generate a report, you could have a check box "add tag to exported records" that you can check and then clicking any button (copy, save...) would add the tag.
- You add a "Batch" button next to the Report button, and this button open a new UI with actions you can do on the currently displayed records, so to edit a tag, you would have to select a time range, click on a tag, click the "Batch" button, select "Add tag" and click "Apply".
